src/Collection/Collection.service.ts

```diff
@@ -142,31 +166,35 @@ export class CollectionService {
         )
       }
 
-      const { third_party_id, urn_suffix } = decodeTPCollectionURN(
-        collectionJSON.urn
-      )
-
       if (
         third_party_id !== collection.third_party_id ||
         urn_suffix !== collection.urn_suffix
       ) {
-        const collectionItems = await thirdPartyAPI.fetchThirdPartyCollectionItems(
-          collection.third_party_id!,
-          collection.urn_suffix!
-        )
         // We can't change the TPW collection's URN if there are already published items
-        if (collectionItems.length > 0) {
-          throw new CollectionAlreadyPublishedException(
-            id,
-            CollectionType.THIRD_PARTY,
-            'updated'
-          )
-        }
+        await this.checkIfThirdPartyCollectionHasPublishedItems(
+          id,
+          collection.third_party_id,
+          collection.urn_suffix
+        )
+
+        // Check if the new URN for the collection already exists
+        await this.checkIfThirdPartyCollectionHasPublishedItems(
+          id,
+          third_party_id,
+          urn_suffix
+        )
```

Comment:
Would it make sense to do this in a Promise.all?

Reply:
That is actually a great question. In theory it shouldn't make a request to the graph with the new id's if the older ones already have items published, so it's a sequential procedure, but we could make it parallelized by requesting that data to the graph anyways. What do you think?